### ISSUE TITLE
Typed arrays exported from tool scripts cannot be edited

### ISSUE THREAD

**Issue by maiself** [godotengine/godot#76642]
### Godot version

9f12e7b52d944

### System information

arch linux

### Issue description

Attempting to edit from the editor a typed array exported from a tool script fails, any value is immediately reset, setters are never called.

### Steps to reproduce

Create a script such as follows and try to add and then change items of the array in the inspector.

```gdscript
@tool
extends Node

@export var x: Array[String]
```

### Minimal reproduction project

n/a

**Comment by kleonc** [comment]
> Reverting the commit from #76378 resolves this issue.

cc @KoBeWi, regressioception. :laughing:

**Comment by KoBeWi** [comment]
No idea why this happens. The text sub-property is re-created for some reason, but only in tool script.

**Comment by jpcerrone** [comment]
This seems to be fixed on the latest version. I tested both on the commit referenced by OP and on (https://github.com/godotengine/godot/commit/8c729f0f34a92b92a1a8204e1d56b6ad05cfd27b). While the issue did appear in the first one, it is no longer present on the latest build.

Tested on Windows 11, Forward+ mode.
![noBugArray](https://user-images.githubusercontent.com/38991758/236213726-9ccd9e54-9ea9-47c3-8b7f-b1f150bf59ee.gif)
